File: backend/app/schemas/discover.py
"""Pydantic schemas for discover listing, detail, and publish body."""

from pydantic import BaseModel, Field


class DiscoverPublishBody(BaseModel):
    """Owner payload when publishing a notebook to the discover catalog."""

    category: str = Field(default="general", max_length=64)
    cover_url: str = Field(default="", max_length=512)


class DiscoverNotebookListItem(BaseModel):
    """One row in the public discover notebook grid."""

    id: str
    title: str
    description: str
    category: str
    cover_url: str
    subscriber_count: int
    source_count: int
    owner_display_name: str


class DiscoverNotebookListResponse(BaseModel):
    """Paginated discover list."""

    items: list[DiscoverNotebookListItem]
    total: int


class DiscoverNotebookDetail(BaseModel):
    """Public detail for a discoverable notebook (DTO for API responses)."""

    id: str
    title: str
    description: str
    category: str
    cover_url: str
    subscriber_count: int
    source_count: int
    owner_display_name: str
    share_token: str | None

File: frontend/src/api/discover.ts
import client from './client'
import { publicOnly } from './publicClient'

export interface DiscoverNotebookListItem {
  id: string
  title: string
  description: string
  category: string
  cover_url: string
  subscriber_count: number
  source_count: number
  owner_display_name: string
}

export interface DiscoverNotebookListResponse {
  items: DiscoverNotebookListItem[]
  total: number
}

export interface DiscoverNotebookDetail extends DiscoverNotebookListItem {
  share_token: string | null
}

export async function fetchDiscoverNotebooks(params: {
  q?: string
  category?: string
  offset?: number
  limit?: number
}): Promise<DiscoverNotebookListResponse> {
  const res = await publicOnly.get<DiscoverNotebookListResponse>(
    '/public/discover/notebooks',
    { params },
  )
  return res.data
}

export async function fetchDiscoverNotebookDetail(
  id: string,
): Promise<DiscoverNotebookDetail> {
  const res = await publicOnly.get<DiscoverNotebookDetail>(
    `/public/discover/notebooks/${id}`,
  )
  return res.data
}

export async function subscribeDiscoverNotebook(id: string): Promise<void> {
  await client.post(`/discover/notebooks/${id}/subscribe`)
}

export async function unsubscribeDiscoverNotebook(id: string): Promise<void> {
  await client.delete(`/discover/notebooks/${id}/subscribe`)
}
